### PR TITLE
define a PR template with acceptance checklist

### PR DESCRIPTION
**Pull request by garypen**
This is to promote conversation.

### CODE REVIEW

**Review by bnjjj — APPROVED** [review]

**Review by abernix — CHANGES_REQUESTED** [review]
I like these changes from the checklist standpoint!  As of right now though, I think this is going to show up as choice in the "New Issue Chooser" [workflow](https://github.com/apollographql/router/issues/new/choose) as "Feature" — for **everyone**, not just the team.

I'm actually not sure if that's the workflow we're trying to enable here and I suspect what we want is a set of acceptance criteria that we are considerate of for every incoming PR — including existing issues that are open and haven't had PRs raised for them yet, or for PRs that are opened without issues.

Perhaps what we need to do is make this list the criteria for opening and reviewing PRs _in a general sense_, whether they are from the core contributors or the greater community.  I believe this could beg for this to be moved into [the PR template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) and something that we should _expect_ (and require) be filled in and checked off for each PR?

**Comment by garypen** [comment]
> I like these changes from the checklist standpoint! As of right now though, I think this is going to show up as choice in the "New Issue Chooser" [workflow](https://github.com/apollographql/router/issues/new/choose) as "Feature" — for **everyone**, not just the team. (Note that _Feature request_ is already there)
> 
> I'm actually not sure if that's the workflow we're trying to enable here and I suspect what we want is a set of acceptance criteria that we are considerate of for every incoming PR — including existing issues that are open and haven't had PRs raised for them yet, or for PRs that are opened without issues. That said, it is certainly plausible that some aspect of this should still be incorporated into the general issue templates?
> 
> Perhaps what we need to do is make this list the criteria for opening and reviewing PRs _in a general sense_, whether they are from the core contributors or the greater community. I believe this could beg for this to be moved into [the PR template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) and something that we should _expect_ (and require) be filled in and checked off for each PR?

I've made this into a PR template and renamed the PR to reflect that.

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+*Note any exceptions here*
+
+**Notes**
+
+1. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
+2. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
+3. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
+    - please document the manual testing (extensively) in the Exceptions.
+    - please raise a separate issue to automate the test (tag: manual test)
```
Comment (abernix) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:32]:
```suggestion
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`
```

**Review by abernix — APPROVED** [review]
LGTM with two points:

1. (Noted on a line review below) A clarification that tries to point out that a contributor (themselves) wouldn't have sufficient permission to label an issue. (And label instead of tag because of GH convention.)

2. If my interpretation of [the docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository#adding-a-pull-request-template) is correct, I think the current choice of file means it **may** require the use of the the `template` query string parameter to specify this file.

	Put another way, it would be the _default_ template if the file name was `.github/pull_request_template.md`, rather than `.github/PULL_REQUEST_TEMPLATE/pull_request_template.md`.
	
	But we can probably test it first using the query string parameter approach, so :shipit: !

**Review by abernix — CHANGES_REQUESTED** [review]
Changed to "Request changes" because I don't understand why this says there are so many commits in this PR.

**Comment by garypen** [comment]
Closing. Replaced by #2355

**Pull request closed without merging**